File: MB-Secretaria/confidence/uncertainty-score.ts
import type { EvidenceEvaluation, HallucinationRisk, ProviderConsistencyResult } from "./types";

export function calculateUncertaintyScore(input: {
  evidence: EvidenceEvaluation;
  providerConsistency: ProviderConsistencyResult;
  hallucinationRisk: HallucinationRisk;
}): number {
  let uncertainty = 1 - input.evidence.evidence_completeness;
  uncertainty += 1 - input.providerConsistency.consistency_score;
  uncertainty /= 2;
  if (input.hallucinationRisk.risk_level === "high") uncertainty += 0.15;
  if (input.hallucinationRisk.risk_level === "critical") uncertainty += 0.25;
  return Number(Math.max(0, Math.min(1, uncertainty)).toFixed(3));
}

File: MB-Secretaria/confidence_py/uncertainty_score.py
from .types import EvidenceEvaluation, HallucinationRisk, ProviderConsistencyResult


def calculate_uncertainty_score(
    evidence: EvidenceEvaluation,
    provider_consistency: ProviderConsistencyResult,
    hallucination_risk: HallucinationRisk,
) -> float:
    uncertainty = ((1 - evidence.evidence_completeness) + (1 - provider_consistency.consistency_score)) / 2
    if hallucination_risk.risk_level == "high":
        uncertainty += 0.15
    if hallucination_risk.risk_level == "critical":
        uncertainty += 0.25
    return round(max(0.0, min(1.0, uncertainty)), 3)
